src/pmpo_utils.hpp

```diff
@@ -210,6 +453,30 @@ double sphericalTriangleArea2(Vec3d &a, Vec3d &b, Vec3d &c, double radius){
 //this is a lazy comparison and shouldn't be relied on beyond simple testing
 bool isEqual(double a, double b, double tol=1e-9);
 
+KOKKOS_INLINE_FUNCTION
+Vec3d xyz_from_lat_lon(double lat, double lon, double r){
+  Vec3d xyz;
+  xyz[0] = r*cos(lat)*cos(lon);
+  xyz[1] = r*cos(lat)*sin(lon);
+  xyz[2] = r*sin(lat);
+  return xyz;
+}
+
+KOKKOS_INLINE_FUNCTION
+Vec3d grid_rotation_backward(Vec3d& xyz_input){
```

Comment:
Need a comment here to indicate specific rotation, e.g., rotation by angle of 90 degrees in a certain way, isn't it?